src/main/java/com/purple/ua/universityappointment/controller/LessonController.java

```diff
@@ -23,9 +30,52 @@
     @Autowired
     private LessonService lessonService;
 
+    @PreAuthorize("hasRole('ROLE_LECTURER')")
+    @GetMapping("/all")
+    ResponseEntity<List<Lesson>> getAllLessons(@AuthenticationPrincipal MyUserDetails userDetails) {
+        List<Lesson> allLessons = lessonService.getAllLesson(userDetails.getUser());
+        return new ResponseEntity(allLessons, HttpStatus.OK);
+    }
+
+    @GetMapping
+    ResponseEntity<LessonDto> getLessonById(@RequestParam long id) {
+        Lesson lesson = lessonService.getLessonById(id);
+        return new ResponseEntity<>(INSTANCE.toDto(lesson, new CycleAvoidingMappingContext()), HttpStatus.OK);
+    }
+
+    @GetMapping("/lecturer")
+    ResponseEntity<List<LessonDto>> getLessonsByLecturerId(@RequestParam long id) {
+        List<Lesson> lessons = lessonService.getLessonsByLecturerId(id);
+        return new ResponseEntity<>(INSTANCE.listToDto(lessons, new CycleAvoidingMappingContext()), HttpStatus.OK);
+    }
+
+    @GetMapping("/lecturer/name")
+    ResponseEntity<List<LessonDto>> getLessonsByLecturerFirstAndLastName(@RequestParam String firstName, @RequestParam String lastName) {
+        List<Lesson> lessons = lessonService.getLessonsByLecturerFirstNameAndLastName(firstName, lastName);
+        return new ResponseEntity<>(INSTANCE.listToDto(lessons, new CycleAvoidingMappingContext()), HttpStatus.OK);
+    }
+
+
+    @PreAuthorize("hasRole('ROLE_LECTURER')")
     @PostMapping("/create")
-    ResponseEntity<LessonDto> createLesson(@Valid @RequestBody LessonDto lessonDto) {
-        lessonService.createLesson(INSTANCE.toEntity(lessonDto));
-        return new ResponseEntity(lessonDto, HttpStatus.CREATED);
+    ResponseEntity<LessonDto> createLesson(@RequestBody LessonDto lessonDto, @AuthenticationPrincipal MyUserDetails userDetails) {
```

Comment:
MyUserDetails looks worth than User)) so, I recommend User to implement UserDetails